cupy/_core/internal.pyx

```diff
@@ -547,3 +545,37 @@ cdef bint _is_layout_expected(
         return True
     else:
         return False
+
+
+cdef bint check_aligned(_ndarray_base arr) except -1:
+    """Check if an array is aligned with its dtype. We usually assume
+    that this is the case, since cuda allocations are aligned.
+
+    NOTE: We currently assume structured dtypes are always aligned for lack
+    of a clearer definition of what it means for them to be aligned.
+    """
+    cdef Py_ssize_t itemsize = arr.dtype.itemsize
+    cdef Py_ssize_t pow_2_mask = itemsize - 1
```

Comment:
This means we check for an alignment to the itemsize.  I am not actually 100% sure that is correct, but I _think_ that cuda will align and assume alignment of itemsize?

I thought briefly to just not check, but this leads to a `CUDARuntimeError` and that may be fine, but it is a sticky error, i.e. once you run into it, you have to restart the whole session.